src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -68,46 +69,142 @@ internal static TupleTypeSymbol ConstructTupleTypeSymbol(TupleTypeSymbol origina
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5__Item5 },
+
+                                                        new[]{
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item1,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item2,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item3,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item4,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item5,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item6 },
+
+                                                        new[]{
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item1,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item2,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item3,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item4,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item5,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item6,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item7 }
+        };
+
+        /// <summary>
+        /// Returns "Item1" for position=0
+        /// </summary>
+        internal static string UnderlyingMemberName(int position)
+        {
+            if (position > 6)
+            {
+                // PROTOTYPE
+                position = 0;
+            }
+            return memberNames[position];
+        }
+
+        /// <summary>
+        /// Returns true for "Item1, "Item2", ..., "Rest"
+        /// </summary>
+        internal static bool IsMemberNameReserved(string name)
+        {
+            // PROTOTYPE handle "Rest" and others like "ToString"?
+            return memberNames.Contains(name, StringComparer.Ordinal);
```

Comment:
I believe Vlad recently added a special ordinal comparer for strings. Should we be using it instead?